graphql/schema.graphql

```diff
@@ -1,34 +1,33 @@
-type Book @autopersist {
-  id: ID!
-  name: String @searchable @cacheControl(maxAge: 120)
-  author: String @cacheControl(maxAge: 120)
-  languages: [String]
-  edition: Edition @searchable @cacheControl(scope: PRIVATE)
-}
+# To really understand GraphQL, go to http://graphql.org/learn/
 
-type Edition {
-  pages: Int
-  number: Int
+# Autopersisted types are stored automatically in VTEX masterdata
+type AutopersistedBook @autopersist {
+  id: ID!                   # Autopersisted types must have an ID
+  name: String @searchable  # @searchable allows you to fetch a book by it's name
+  authors: [String!]        # This is an array of non-null strings
 }
 
-type Order @autopersist(entity:  "orders") {
+# Cache control can be set at each field
+# ref: https://www.apollographql.com/docs/engine/caching.html
+type CachedBook @cacheControl(scope: PUBLIC, maxAge: 120) {
   id: ID!
-  status: String  @cacheControl(scope: PRIVATE)
-  statusDescription: String @cacheControl(scope: PRIVATE)
-  orderedBy: User! @cacheControl(maxAge: 120)
+  name: String       @cacheControl(scope: PUBLIC, maxAge: 40)
+  authors: [String!] @cacheControl(scope: PRIVATE)
 }
 
-type User {
-  id: ID!
+# It is necessary to generate input types for creating books
+# ref: http://graphql.org/graphql-js/mutations-and-input-types/
+input cachedBookInput {
```

Comment:
Types in GraphQL usually start with an uppercase letter, even input types.

Reply:
Thanks 👍